src/screens/Departures/state/stop-place-state.ts

```diff
@@ -278,15 +286,15 @@ export function useStopPlaceData(
       }),
     [stopPlace.id, favoriteDepartures, showOnlyFavorites],
   );
-  useEffect(refresh, [stopPlace.id, startTime]);
+  useEffect(dispatchLoadInitialDepartures, [stopPlace.id, startTime]);
   useEffect(() => {
     if (!state.tick) {
       return;
     }
     const diff = differenceInMinutes(state.tick, state.lastRefreshTime);
 
     if (diff >= HARD_REFRESH_LIMIT_IN_MINUTES) {
-      refresh();
+      dispatchLoadInitialDepartures();
```

Comment:
here i mean :)

Reply:
Based on our discussion then I see that using the refresh callback in the useEffect is Ok as well, so I have reverted my changes.